src/v/iceberg/datatypes_json.cc

```diff
@@ -177,6 +177,10 @@ void rjson_serialize(iceberg::json_writer& w, const iceberg::nested_field& f) {
     w.Bool(bool(f.required));
     w.Key("type");
     rjson_serialize(w, f.type);
+    w.Key("initial-default");
+    w.Null();
+    w.Key("write-default");
```

Comment:
Maybe leave a TODO explaining that this isn't a bug to always serialize null, and that we just don't support required fields yet?

Though the `redpanda` struct has some required fields -- is it invalid to serialize nulls as their defaults?

Reply:
good points. rereading the spec:

>The initial-default is set only when a field is added to an existing schema. The write-default is initially set to the same value as initial-default and can be changed through schema evolution. If either default is not set for an optional field, then the default value is null for compatibility with older spec versions.

suggests that this isn't required at all. so maybe the correct thing is to

- omit the `*-default` fields
- leave a TODO to the effect of "when/if we add support for required fields, 'initial-default' and 'write-default' should be written here" or something like that

Reply:
done, as described above